shards/modules/physics/soft_body.cpp

```diff
@@ -0,0 +1,422 @@
+
+    if (!_rotation.isVariable())
+      throw std::runtime_error("Rotation must be a variable");
+    auto rv = findExposedVariable(data.shared, _rotation);
+    if (!rv || !rv->isMutable)
+      throw std::runtime_error("Rotation must be mutable");
+
+    if (!_shape.isVariable())
+      throw std::runtime_error("Shape must be set");
+
+    // Determine dynamic parameters
+    if (_friction.isVariable())
+      _dynamicParameterMask |= SBPTM_Friction;
+    if (_restitution.isVariable())
+      _dynamicParameterMask |= SBPTM_Restitution;
+    if (_pressure.isVariable())
+      _dynamicParameterMask |= SBPTM_Pressure;
+    if (_linearDamping.isVariable())
+      _dynamicParameterMask |= SBPTM_LinearDamping;
+    if (_maxLinearVelocity.isVariable())
+      _dynamicParameterMask |= SBPTM_MaxLinearVelocity;
+    if (_gravityFactor.isVariable())
+      _dynamicParameterMask |= SBPTM_GravityFactor;
+    if (_collisionGroup.isVariable())
+      _dynamicParameterMask |= SBPTM_CollisionGroup;
+
+    return outputTypes().elements[0];
+  }
+
+  JPH::EAllowedDOFs convertAllowedDOFs(const SHVar &input) {
+    if (input.valueType != SHType::Seq) {
+      return (JPH::EAllowedDOFs)input.payload.enumValue;
+    } else {
+      JPH::EAllowedDOFs dofs = JPH::EAllowedDOFs::None;
+      for (auto &dof : input) {
+        dofs |= (JPH::EAllowedDOFs)dof.payload.enumValue;
+      }
+      return dofs;
+    }
+  }
+
+  void initNewNode(std::shared_ptr<BodyNode> node) {
+    if ((bool)*_static)
+      node->persistence = true;
+    node->enabled = _enabled.get().payload.boolValue;
+    node->tag = _tag.get();
+
+    auto &outParams = node->params.soft;
+    outParams.friction = _friction.get().payload.floatValue;
+    outParams.restitution = _restitution.get().payload.floatValue;
+    outParams.pressure = _pressure.get().payload.floatValue;
+    outParams.linearDamping = _linearDamping.get().payload.floatValue;
+
+    outParams.maxLinearVelocity = _maxLinearVelocity.get().payload.floatValue;
+    outParams.gravityFactor = _gravityFactor.get().payload.floatValue;
+
+    auto cm = toUInt2(_collisionGroup.get());
+    outParams.groupMembership = cm.x;
+    outParams.collisionMask = cm.y;
+
+    node->location = toJPHVec3(_location.get().payload.float3Value);
+    node->rotation = toJPHQuat(_rotation.get().payload.float4Value);
+    node->updateParamHash0();
+    node->updateParamHash1();
+  }
+
+  SHVar activate(SHContext *shContext, const SHVar &input) {
+    ZoneScopedN("Physics::entity");
+
+    if (!_instance->node) {
+      _instance->node = _requiredContext->core->createNewBodyNode();
+      _instance->core = _requiredContext->core;
+      initNewNode(_instance->node);
+    }
+
+    auto &instNode = _instance->node;
+
+    instNode->enabled = _enabled.get().payload.boolValue;
+    _requiredContext->core->touchNode(instNode);
+
+    // Filtering to avoid unnecessary checks for static parameters
+    auto &outParams = instNode->params.soft;
```

Comment:
The activate method in SoftBodyShard checks dynamic parameters individually. Consider using a bitwise AND operation for potentially faster checks.

Reply:
Don't I already do that :smile: ?

Reply:
I think something could be aggregated. Didn't dig too deep tho.

Reply:
Compiler probably does anyway